packages/sst/src/constructs/RemixSite.ts

```diff
@@ -66,6 +71,13 @@ export class RemixSite extends SsrSite {
       },
     };
 
+    // The path for all files that need to be in the "/" directory (static assets)
+    // is different when using Vite. These will be located in the "build/client"
+    // path of the output. It will be the "public" folder when using remix config.
+    const publicAssetsPath = this.usingVite
+      ? path.join("build", "client")
```

Comment:
Hardcoding these values makes it extremely difficult to adopt to new defaults that Remix might choose to introduce in the future.

Perhaps introducing some configurables on the RemixSite construct might be a better way. That way we don't have to care whether it's Vite or not as all SST needs to know is where to read from.

As for the `*-server.js` support files, it might be better to treat them as templates as they are not executed until they are in the lambda. That way in combination with aforementioned configurables it can be easier to control where the import (`import * as remixServerBuild from "./server/index.js";`) happens from.

Reply:
hmm.. in Astro, there's this adapter concept. And the `astro-sst` adapter writes all the build config (ie. assets build, server build) to a manifest file, and the `AstroSite` construct then read from the manifest file.

Next.js also generates these manifest files.

Is there something similar for Remix?

Reply:
For the templates we can probably just use the `esbuild: inject` where I had removed the polyfills. From there we can just inject the single line of where to import the server build from. This will allow us to have a single template and dynamically specify where the actually server entry is located.

Also hardcoding is what we do currently, if you check the code when we examine for the `remix.config.js` file we ensure that these values are either implicit or hardcoded exactly. I do agree giving more of the options would be good, but thinking maybe that should come in a separate PR. At least we could add Vite support for time being. I actually believe we will need a little bit of a rewrite as Remix offers the `serverBranchBundles` concept now - this allow splitting the app into chunks for multiple lambdas. I don’t believe the `SsrSite` itself allows this, IMO we should consider expanding the config and including this detail at the same time.

Reply:
So to clarify on the inject:
1. We can remove the import from the template.
2. We can check is using Vite, and then inject a line of code that does the `import * as remixServerBuild` in the `createServerLambdaBundle` function.

Reply:
> hmm.. in Astro, there's this adapter concept. And the `astro-sst` adapter writes all the build config (ie. assets build, server build) to a manifest file, and the `AstroSite` construct then read from the manifest file.
> 
> Next.js also generates these manifest files.
> 
> Is there something similar for Remix?

Will check for manifest files tonight after getting back from office and see if this is possible. This would solve the biggest hurdle of not being able to dynamically import the `vite.config.ts` file, we also can’t use require because it’s not guaranteed to be written in CJS format.

Reply:
> splitting the app into chunks for multiple lambdas
In `plan()`, you can define multiple server functions and define which behaviors use which function. None of the SSR site constructs is doing it.. but should be supported. But agreed that should be a separate PR.

Reply:
> > splitting the app into chunks for multiple lambdas
> 
> In `plan()`, you can define multiple server functions and define which behaviors use which function. None of the SSR site constructs is doing it.. but should be supported. But agreed that should be a separate PR.

Oh terrific, did not know this! Will have to investigate for a PR on this after this one.

Reply:
Regarding the manifest file, starting with [Remix 2.6.0](https://github.com/remix-run/remix/blob/main/CHANGELOG.md#v260) it respects the explicit setting in `vite.config.ts` for generating the manifest.

Reply:
Ok have gotten deploys working and removed the two added `support/` files. Only had to change the `createServerLambdaBundle` but am now doing the injection in the procedure itself. Probably fine, but would like an opinion if you guys think this looks good for this PR?

(Section that changed is the `buildInjector` section.)

```ts
  private createServerLambdaBundle(wrapperFile: string) {
    // Create a Lambda@Edge handler for the Remix server bundle.
    //
    // Note: Remix does perform their own internal ESBuild process, but it
    // doesn't bundle 3rd party dependencies by default. In the interest of
    // keeping deployments seamless for users we will create a server bundle
    // with all dependencies included. We will still need to consider how to
    // address any need for external dependencies, although I think we should
    // possibly consider this at a later date.

    // In this path we are assuming that the Remix build only outputs the
    // "core server build". We can safely assume this as we have guarded the
    // remix.config.js to ensure it matches our expectations for the build
    // configuration.
    // We need to ensure that the "core server build" is wrapped with an
    // appropriate Lambda@Edge handler. We will utilise an internal asset
    // template to create this wrapper within the "core server build" output
    // directory.

    // Ensure build directory exists
    const buildPath = path.join(this.props.path, "build");
    fs.mkdirSync(buildPath, { recursive: true });

    // Copy the server lambda handler and pre-append the build injection based
    // on the config file used.
    //
    // Note: When using Vite config, the output build will be "server/index.js"
    // and when using Remix config it will be `server.js`.
    const buildInjector = [
      `// We have to ensure that our polyfills are imported prior to any other modules`,
      `// which may depend on them`,
      `import { installGlobals } from "@remix-run/node";`,
      `installGlobals();`,
      ``,
      `// Import the server build that was produced by 'remix build'`,
      `import * as remixServerBuild from "${
        this.usingVite ? "./server/index.js" : "./server.js"
      }";`,
      ``,
    ].join("\n");
    fs.writeFileSync(path.resolve(buildPath, "server.js"), buildInjector);
    const wrapperFileContents = fs.readFileSync(
      path.resolve(__dirname, `../support/remix-site-function/${wrapperFile}`),
      { encoding: "utf8" }
    );
    fs.appendFileSync(
      path.resolve(buildPath, "server.js"),
      wrapperFileContents
    );

    // Copy the Remix polyfil to the server build directory
    //
    // Note: We need to ensure that the polyfills are injected above other code that
    // will depend on them when not using Vite. Importing them within the top of the
    // lambda code doesn't appear to guarantee this, we therefore leverage ESBUild's
    // `inject` option to ensure that the polyfills are injected at the top of
    // the bundle.
    const polyfillDest = path.join(buildPath, "polyfill.js");
    fs.copyFileSync(
      path.resolve(__dirname, "../support/remix-site-function/polyfill.js"),
      polyfillDest
    );

    return {
      handler: path.join(buildPath, "server.handler"),
      inject: this.usingVite ? [] : [polyfillDest],
    };
  }
```